### ISSUE TITLE
Can I customize the generated HTML markup?

### ISSUE THREAD

**Issue by yurifyodorov** [tailwindlabs/headlessui#185]
I use Cypress for writing automated tests. 
And there is an option to select `<option>` from the `<select>` element in such a simple way:

```html
<select>
  <option value="456">apples</option>
  <option value="457">oranges</option>
  <option value="458">bananas</option>
</select>
```

```javascript
cy.get('select')
  .select('456').should('have.value', '456')
```

But, unfortunately, there is [one rule](https://docs.cypress.io/api/commands/select.html#Requirements). 

To select option in this way, the elements must be semantically correct (i.e. strictly `<select>` and `<option>`)

In Headless UI, tags are used for lists ( `<ul>` and `<li>` ).

![headless-ui-option](https://user-images.githubusercontent.com/33720649/103092446-4cf8f980-4608-11eb-9418-6a62ead66c63.png)

Is it possible to somehow configure which tag should be used?

**Comment by MrCrayon** [comment]
There are reasons why `select` and `option` are not used but you can still do something like:
```javascript
cy.get(`[data-test="your-select"]`).within(() => {
  cy.get('button').click();
  cy.contains(optionLabel).click();
})
```
Or any other way to select the option you want.

**Comment by yurifyodorov** [comment]
@MrCrayon You can of course and so. Thanks 

I usually do, I just wanted to use the native Cypress methods to simplify the tests.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

You can use the `as` property to change the underlying html structure if you want. However, do note that we explicitly *don't* use `select` because you can't style it and you can't style the `<option>` tags either. Therefore we use `ul` and `li` elements.

If you want to use `select` and `option`, then you can't style the elements. So if you really want to use those, then you don't have to use the Headless UI component.

Long story short, if you want to use select/option so that you can use them in cypress -> don't use Headless UI. Want to style your options, then you would have to update the cypress code.

**Comment by yurifyodorov** [comment]
@RobinMalfait Thank you for your answer!
